src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,81 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String OPERATION_VALUE_DELIMITER = ",";
+    private static final String SUPPLY_OPERATION_TITLE = "supply";
+    private static final String BUY_OPERATION_TITLE = "buy";
+    private static final String RESULT_OPERATION_TITLE = "result";
+    private static final int OPERATION_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Table table = createTableWithTotals(readLinesFromFileToArray(fromFileName));
+        writeFile(composeReport(table.getSupply(), table.getBuy(), table.calculateResult()),
+                toFileName);
+    }
+
+    private String[] readLinesFromFileToArray(String filename) {
+        StringBuilder linesFromFile = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filename))) {
+            String string = bufferedReader.readLine();
+            while (string != null) {
+                linesFromFile.append(string).append(System.lineSeparator());
+                string = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("File not found or can't read from file " + filename, e);
+        }
+        return String.valueOf(linesFromFile).split(System.lineSeparator());
+    }
+
+    private Table createTableWithTotals(String[] lines) {
+        int buy = 0;
+        int supply = 0;
+        for (String operation : lines) {
+            String[] operationData = operation.split(OPERATION_VALUE_DELIMITER);
+            switch (operationData[OPERATION_INDEX]) {
+                case BUY_OPERATION_TITLE: {
+                    buy += Integer.parseInt(operationData[VALUE_INDEX]);
+                    break;
+                }
+                case SUPPLY_OPERATION_TITLE: {
+                    supply += Integer.parseInt(operationData[VALUE_INDEX]);
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+        return new Table(buy, supply);
+    }
+
+    private String composeReport(int supplyAmount, int buyAmount, int result) {
```

Comment:
```suggestion
    private String composeReport(Table table) {
```

create local variables for supplyAmount, buyAmount, result